processor.go

```diff
@@ -314,7 +284,7 @@ func (g *Processor) Run(ctx context.Context) (rerr error) {
 		g.log.Debugf("closing producer")
 		defer g.log.Debugf("producer ... closed")
 		if err := g.producer.Close(); err != nil {
-			merrors.Collect(fmt.Errorf("error closing producer: %v", err))
+			errs = multierror.Append(errs, fmt.Errorf("error closing producer: %w", err))
```

Comment:
Setting the `errs` here in this deferred func and then later using the `errs` to fill the `rerr` in the previous (but later called) deferred func still twists my head. This whole reversed defer processing makes this really hard to understand.
I think the code is correct, but if there is a cleaner way to write this, the code would be more maintainable.

Reply:
I also don't like to modify the the return value in a deferred function, but haven't found a better solution yet. Maybe we'll find another way.